subsys/suit/storage/src/suit_storage_nrf9280.c

```diff
@@ -0,0 +1,1085 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <suit_storage_mpi.h>
+#include <suit_storage_nvv.h>
+#include <suit_storage_internal.h>
+#include <zephyr/logging/log.h>
+#include <psa/crypto.h>
+
+LOG_MODULE_REGISTER(suit_storage, CONFIG_SUIT_LOG_LEVEL);
+
+/* SUIT storage partition is defined as reserved-memory region, thus it uses absolute addresses. */
+#define SUIT_STORAGE_OFFSET                                                                        \
+	(FIXED_PARTITION_OFFSET(suit_storage_partition) - DT_REG_ADDR(DT_CHOSEN(zephyr_flash)))
+#define SUIT_STORAGE_SIZE		 FIXED_PARTITION_SIZE(suit_storage_partition)
+#define SUIT_STORAGE_NORDIC_ADDRESS	 suit_plat_mem_nvm_ptr_get(SUIT_STORAGE_NORDIC_OFFSET)
+#define SUIT_STORAGE_NORDIC_OFFSET	 FIXED_PARTITION_OFFSET(cpusec_suit_storage)
+#define SUIT_STORAGE_NORDIC_SIZE	 FIXED_PARTITION_SIZE(cpusec_suit_storage)
+#define SUIT_STORAGE_NORDIC_CELL_ADDRESS suit_plat_mem_nvm_ptr_get(SUIT_STORAGE_NORDIC_CELL_OFFSET)
+#define SUIT_STORAGE_NORDIC_CELL_OFFSET  FIXED_PARTITION_OFFSET(cpucell_suit_storage)
+#define SUIT_STORAGE_NORDIC_CELL_SIZE    FIXED_PARTITION_SIZE(cpucell_suit_storage)
+#define SUIT_STORAGE_RAD_ADDRESS	 suit_plat_mem_nvm_ptr_get(SUIT_STORAGE_RAD_OFFSET)
+#define SUIT_STORAGE_RAD_OFFSET		 FIXED_PARTITION_OFFSET(cpurad_suit_storage)
+#define SUIT_STORAGE_RAD_SIZE		 FIXED_PARTITION_SIZE(cpurad_suit_storage)
+#define SUIT_STORAGE_APP_ADDRESS	 suit_plat_mem_nvm_ptr_get(SUIT_STORAGE_APP_OFFSET)
+#define SUIT_STORAGE_APP_OFFSET		 FIXED_PARTITION_OFFSET(cpuapp_suit_storage)
+#define SUIT_STORAGE_APP_SIZE		 FIXED_PARTITION_SIZE(cpuapp_suit_storage)
+
+BUILD_ASSERT((SUIT_STORAGE_OFFSET <= SUIT_STORAGE_NORDIC_OFFSET) &&
+		     (SUIT_STORAGE_NORDIC_OFFSET + SUIT_STORAGE_NORDIC_SIZE <=
+		      SUIT_STORAGE_OFFSET + SUIT_STORAGE_SIZE),
+	     "Secure storage must be defined within SUIT storage partition");
+
+BUILD_ASSERT((SUIT_STORAGE_OFFSET <= SUIT_STORAGE_RAD_OFFSET) &&
+		     (SUIT_STORAGE_RAD_OFFSET + SUIT_STORAGE_RAD_SIZE <=
+		      SUIT_STORAGE_OFFSET + SUIT_STORAGE_SIZE),
+	     "Radiocore storage must be defined within SUIT storage partition");
+
+BUILD_ASSERT((SUIT_STORAGE_OFFSET <= SUIT_STORAGE_APP_OFFSET) &&
```

Comment:
You may also extend those asserts to cover `SUIT_STORAGE_NORDIC_CELL_OFFSET` and `SUIT_STORAGE_NORDIC_CELL_SIZE` too.

Reply:
Fixed.